packages/alfa-rules/src/sia-r84/rule.ts

```diff
@@ -0,0 +1,97 @@
+import { Rule } from "@siteimprove/alfa-act";
+import { Device } from "@siteimprove/alfa-device";
+import { Element, Namespace } from "@siteimprove/alfa-dom";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Ok, Err } from "@siteimprove/alfa-result";
+import { Style } from "@siteimprove/alfa-style";
+import { Page } from "@siteimprove/alfa-web";
+
+import { expectation } from "../common/expectation";
+
+import { hasDescendant } from "../common/predicate/has-descendant";
+import { isVisible } from "../common/predicate/is-visible";
+import { isTabbable } from "../common/predicate/is-tabbable";
+
+const { isElement, hasNamespace } = Element;
+const { and, or } = Predicate;
+
+export default Rule.Atomic.of<Page, Element>({
+  uri: "https://siteimprove.github.io/sanshikan/rules/sia-r84.html",
+  evaluate({ device, document }) {
+    return {
+      applicability() {
+        return document
+          .descendants({ flattened: true, nested: true })
+          .filter(
+            and(
+              isElement,
+              and(
+                hasNamespace(Namespace.HTML),
+                isVisible(device),
+                isPossiblyScrollable(device)
+              )
+            )
+          );
+      },
+
+      expectations(target) {
+        return {
+          1: expectation(
+            or(
+              isTabbable(device),
+              hasDescendant(and(isElement, isTabbable(device)), {
+                flattened: true,
+              })
```

Comment:
Do we have a notion of inclusive descendant (https://dom.spec.whatwg.org/#concept-tree-inclusive-descendant)? Should we? (maybe as an option for `hasDescendant`)

Reply:
That's a great point! I definitely think we should 👍